### ISSUE TITLE
Speed up BTCPay Redirect Ideas

### ISSUE THREAD

**Issue by jaxkeiser** [mynodebtc/mynode#724]
I setup BTCPay Server with myNode for my Woocommerce store using this guide: https://mynodebtc.github.io/btcpay-server/setup.html and I got the https certificate using lets encrypt.

I'm wondering all the ways in which I can increase the speed that my ecommerce store redirects customers to my BTCPay Server checkout screen when customers click pay with BTCPay.

For example, I was able to drastically increase the speed by disabling Tor and I2P for my node within the myNode control panel.

Is there anything else?

**Comment by tehelsper** [comment]
Interesting. I'm not sure how Tor and I2P would make BTCPay go faster, but it could free up some resources.

I would guess the biggest variables are your Internet upload speed and the device. Upgrading those would both make it more responsive, but I'm not sure of your exact bottleneck.

**Comment by jaxkeiser** [comment]
@tehelsper how should I upgrade the device? more RAM?

**Comment by tehelsper** [comment]
What device are you using now? This is sort of outside the scope of mynode, but if a website is not responding quickly it's often sue to internet speed or server speed. Are you using a Raspi 4? How much RAM?

For this setup, I'd probably run myNode on a mini PC (similar to the model two) or as a VM.

**Comment by jaxkeiser** [comment]
I'm using the myNode branded raspberry pi with 4gb of ram, if I updated it to 8gb do you think that would solve any server speed insufficiencies?

**Comment by tehelsper** [comment]
It would help, but upgrading to a more powerful CPU would help as well. That still isn't a guarantee though since running a home server has a lot of variables when it comes to performance.

**Comment by jaxkeiser** [comment]
@tehelsper I didn't open this issue as a bug. I opened it as a "blank issue". I got Zeus wallet connected through Tailscale finally but every time I open the app it has to connect which takes a cartoonishly long amount of time, and then it tries to fetch my balance but times out before it is capable of doing so. During set up I didn't provide a TLS certificate or macaroon. Would providing these things speed up the connection time by any chance? I don't see how the average person is going to be able to use lightning in a non custodial manner with set up being so difficult and connection speed being too slow to be practical enough to whip out your phone, open your wallet and buy a cup of coffee. I'm not trying to complain I'm just trying to give my 2 sats here.
